src/python/WMCore/ReqMgr/Web/ReqMgrService.py

```diff
@@ -521,10 +524,13 @@ def fetch(self, rid):
                 transitions.remove(status)
             visible_attrs = get_modifiable_properties(status)
             filterout_attrs = get_protected_properties()
-
             # extend filterout list with "RequestStatus" since it is passed separately
             filterout_attrs.append("RequestStatus")
 
+            for key, val in self.assignArgs.items():
+                if key not in doc or not doc.get(key):
```

Comment:
Oh yes! I oversaw the first logic in the if. It will be `if not doc.get(key)`, which is the same as you suggested (None is returned by default in case key is not there).

And what I want is to go into the `if ` for cases like `''`, `[]`, `None`, `{}`. Such that we set it to the assignment default (which is going to be the same for most of the cases). I'm updating this patch soon.